trio/_core/_io_epoll.py

```diff
@@ -1,23 +1,38 @@
+from __future__ import annotations
+
 import select
 import sys
 from collections import defaultdict
-from typing import TYPE_CHECKING, Dict
+from typing import TYPE_CHECKING, DefaultDict
 
 import attr
 
 from .. import _core
 from ._io_common import wake_all
-from ._run import _public
+from ._run import Task, _public
 from ._wakeup_socketpair import WakeupSocketpair
 
+if TYPE_CHECKING:
+    from socket import socket
+
+    from .._core import Abort, RaiseCancelT
+
+
+@attr.s(slots=True, eq=False)
+class EpollWaiters:
+    read_task: Task | None = attr.ib(default=None)
+    write_task: Task | None = attr.ib(default=None)
+    current_flags: int = attr.ib(default=0)
+
+
 assert not TYPE_CHECKING or sys.platform == "linux"
 
 
 @attr.s(slots=True, eq=False, frozen=True)
 class _EpollStatistics:
-    tasks_waiting_read = attr.ib()
-    tasks_waiting_write = attr.ib()
-    backend = attr.ib(default="epoll")
+    tasks_waiting_read: int = attr.ib()
+    tasks_waiting_write: int = attr.ib()
+    backend: str = attr.ib(default="epoll")
```

Comment:
Perhaps we might want to make this a `Literal` (and the same with `kqueue`/`windows`), since then type checkers could understand narrowing checks using this? Maybe also change it to a final/classvar.

Reply:
Ah yes, works well with the `Protocol` in #2733.
How do you do class variables with an `attr` class, just remove `attr.ib`?

Reply:
I suppose one could also do a hard-coded property or something

Reply:
I was thinking actually to do a 1-value literal, so the type checker knows that `if stats.backend == "epoll"` will fail for the others. For class variables, `attrs` is smart enough to detect `ClassVar` annotations and skip them.